src/bun.js/api/server.zig

```diff
@@ -1812,16 +1812,28 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
             server.request_pool_allocator.put(this);
         }
 
+        const headers_to_exclude = brk: {
+            if (ssl_enabled) {
+                break :brk [_]JSC.FetchHeaders.HTTPHeaderName{
+                    .ContentLength,
+                    .TransferEncoding,
+                };
+            }
+
+            break :brk [_]JSC.FetchHeaders.HTTPHeaderName{
+                .ContentLength,
+                .TransferEncoding,
+                .StrictTransportSecurity,
+            };
+        };
+
         fn writeHeaders(
             this: *RequestContext,
             headers: *JSC.FetchHeaders,
         ) void {
             ctxLog("writeHeaders", .{});
-            headers.fastRemove(.ContentLength);
-            headers.fastRemove(.TransferEncoding);
-            if (!ssl_enabled) headers.fastRemove(.StrictTransportSecurity);
             if (this.resp) |resp| {
-                headers.toUWSResponse(ssl_enabled, resp);
+                headers.toUWSResponse(ssl_enabled, resp, &headers_to_exclude);
```

Comment:
How is fastRemove fast?

Reply:
It uses headers from this list, which skip storing the original name and skip some of the normalization normally necessary 

https://github.com/zackradisic/bun/blob/main/src/bun.js/bindings/webcore/HTTPHeaderNames.h#L36